lib/src/facts/linux/networking_resolver.cc

```diff
@@ -142,6 +142,17 @@ namespace facter { namespace facts { namespace linux {
         auto parse_route_line = [&known_route_types](string& line, std::vector<route>& routes) {
             vector<boost::iterator_range<string::iterator>> parts;
             boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+
+            // remove trailing "onlink" or "pervasive" flags
+            while (parts.size() > 0) {
```

Comment:
I would've gone for something more like `boost::algorithm::trim_right_if(parts, [](string &s) { return s == "onlink" || s == "pervasive"; });`.